core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -455,6 +456,12 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
                 .toArray(String[]::new);
         createCommand.withEnv(envArray);
 
+        if (binds.size() > 0 && !TestcontainersConfiguration.getInstance().isDisableChecks()) {
+            if (!DockerClientFactory.instance().isSupportsFileMounting()) {
+                VisibleAssertions.warn("Misconfigured file mounting. Some features might not work.");
```

Comment:
Maybe we could be very specific in the warning; something like: `Unable to mount a file from test host into a running container. This may be a misconfiguration or limitation of your Docker environment. Some features might not work.`

Additionally, is it right for this to be a warning, or should we throw an exception? If we're here, it's because the user has tried to do something that will not work, right? Or is it possible for this check to fail but the user's requested FS binds to work?

Reply:
If user is mounting `/var/run/docker.sock`, it will still work, this is why I decided to use warn :)